vocabularies.md

```diff
@@ -0,0 +1,283 @@
+
+All the endpoints here described are restricted to authenticated users.
+
+## Main Endpoint
+**/api/submission/vocabularies**   
+
+Provide access to the configured controlled vocabularies. It returns the list of existent controlled vocabularies.
+
+Example:
+```json
+{
+  "_embedded": {
+    "vocabularies": [
+      {
+        "id": "srsc",
+        "name": "srsc",
+        "scrollable": false,
+        "hierarchical": true,
+        "preloadLevel": 1,
+        "type": "vocabulary",
+        "_links": {
+          "entries": {
+            "href": "https://dspace7.4science.cloud/server/api/submission/vocabularies/srsc/entries"
+          },
+          "self": {
+            "href": "https://dspace7.4science.cloud/server/api/submission/vocabularies/srsc"
+          }
+        }
+      },
+      {
+        "id": "common_types",
+        "name": "common_types",
+        "scrollable": true,
+        "hierarchical": false,
+        "type": "vocabulary",
+        "_links": {
+          "entries": {
+            "href": "https://dspace7.4science.cloud/server/api/submission/vocabularies/srsc/entries"
+          },
+          "self": {
+            "href": "https://dspace7.4science.cloud/server/api/submission/vocabularies/srsc"
+          }
+        }
+      },
+      {
+        "id": "common_iso_languages",
+        "name": "common_iso_languages",
+        "scrollable": true,
+        "hierarchical": false,
+        "type": "vocabulary",
+        "_links": {
+          "entries": {
+            "href": "https://dspace7.4science.cloud/server/api/submission/vocabularies/common_iso_languages/entries"
+          },
+          "self": {
+            "href": "https://dspace7.4science.cloud/server/api/submission/vocabularies/common_iso_languages"
+          }
+        }
+      }
+    ]
+  },
+  "_links": {
+    "self": {
+      "href": "https://dspace7.4science.cloud/server/api/submission/vocabularies"
+    }
+  },
+  "page": {
+    "size": 20,
+    "totalElements": 3,
+    "totalPages": 1,
+    "number": 0
+  }
+}
+```
+
+## Single Vocabulary
+**/api/submission/vocabularies/<:vocabulary-name>**
+
+Provide detailed information about a specific controlled vocabulary. The JSON response document is as follow
+```json
+{
+  "id": "srsc",
+  "name": "srsc",
+  "scrollable": false,
+  "hierarchical": true,
+  "preloadLevel": 1,
+  "type": "vocabulary"
+}
+```
+
+Attributes
+* id: the id of the vocabulary, it is the same than the name
+* name: see id
+* scrollable: if true mean that it is possible to scroll all the entries in the authority without providing a filter parameter, see (vocabulary entries)[vocabularies.md#vocabulary-entries]
+* hierarchical: if true means that the vocabulary expose a tree structure where some entries are parent of others
+* preloadLevel: for hierarchical vocabularies express the preference to preload the tree at a specific level of depth (0 only the top nodes are shown, 1 also their children are preloaded and so on)
```

Comment:
how is this intended to be used? This doesn't exist in DSpace 6

Reply:
it will be used by the Angular UI to open an hierarchical vocabulary with a number of level pre-expanded. This was highly demanded by users in an internal UX test. It is not strictly necessary so if needed we can remove that from the contract and the implementation (that already exists, see screens here https://github.com/DSpace/DSpace/pull/2743#issuecomment-625255712

Reply:
so if I understand it correctly, there are requests for this, and screenshots, but no implementation or config yet?
In that case I think it's best to remove it since we don't want to add more features making the config even more complicated

Reply:
the implementation is already here because we have implemented it firstly for DSpace-CRIS and in https://jira.lyrasis.org/browse/DS-4475 we have noted that explicitly as this allow to keep the effort for DSpace 7 limited.

About the configuration it is just a single line, with a proper default, similar to existing configuration for the DSpaceControlledVocabulary

Reply:
I can see in https://github.com/DSpace/DSpace/pull/2743/files#diff-a60e58808bf42c545ed0dc04a4d6d5f9 there's indeed an open PR to get this in
I'll leave this for @tdonohue whether we allow more features outside the DSpace 6 scope

Reply:
I'll allow this into the contract (especially since an implementation PR exists as of now in https://github.com/DSpace/DSpace/pull/2792).  But, I'll want to review the implementation to see the impact.  I'm generally OK with it though, as it seems small.